### PR TITLE
Clean-up table creation, add indexing to mssql

### PR DESCRIPTION
**Pull request by Arkatufus**
## Changes

- Clean-up table creation, use Linq2Db "create if not exist" flag
- Add indexing to SqlServer journal table

### CODE REVIEW

File: src/Akka.Persistence.Sql.HelperLib/TagTableMigrator.cs [hunk 1]
```diff
@@ -52,10 +52,7 @@ public async Task Migrate(long startOffset, int batchSize, long? endOffset = nul
             await using var connection = _connectionFactory.GetConnection();
 
             // Create the tag table if it doesn't exist
-            var dbSchema = connection.GetSchema();
-
-            if (dbSchema.Tables.All(t => t.TableName != _journalConfig.TableConfig.TagTable.Name))
-                await connection.CreateTableAsync<JournalTagRow>();
+            await connection.CreateTableAsync<JournalTagRow>(TableOptions.CreateIfNotExists);
```
Comment (Aaronontheweb) [inline src/Akka.Persistence.Sql.HelperLib/TagTableMigrator.cs:55]: TagTable name is inside the `TableOptions`, correct?

**Review by Aaronontheweb — COMMENTED** [review]
Left a question

File: src/Akka.Persistence.Sql.HelperLib/TagTableMigrator.cs (hunk 1 above)
Comment (Arkatufus) [reply src/Akka.Persistence.Sql.HelperLib/TagTableMigrator.cs:55]: No, `TagTable` name is inside the table mapping, this is moving the responsibility for detecting if the table exist or not from our code to the database.

Comment (Aaronontheweb) [reply src/Akka.Persistence.Sql.HelperLib/TagTableMigrator.cs:55]: Ah I see - ok, makes sense. Just wanted to make sure I understood the significance of this change. So we're just delegating it to Linq2Db now. Got it.

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**